File: src/model/document.ts
import { DocumentType } from "./document.type";

export class Document {
  private type: DocumentType | null;
  private id: string | null;

  constructor() {
    this.type = null;
    this.id = null;
  }

  getType(): DocumentType | null {
    return this.type;
  }

  getId(): string | null {
    return this.id;
  }

  setType(type: DocumentType): void {
    this.type = type;
  }

  setId(id: string): void {
    this.id = id;
  }
}

File: src/model/Document.java
package model;

import java.util.Objects;

public class Document {
    private DocumentType type;
    private String id;

    public Document () {
    }

    public Document(DocumentType type, String field) {
        this.type = type;
        this.id = field;
    }

    public DocumentType getType() {
        return type;
    }

    public void setType(DocumentType type) {
        this.type = type;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Document document)) return false;
        return getType() == document.getType() && Objects.equals(getId(), document.getId());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getType(), getId());
    }
}
